variables.tf

```diff
@@ -1,36 +1,214 @@
+  default     = 1
+  nullable    = false
 }
 
-variable "plan" {
+variable "dsc_namespace" {
   type        = string
-  description = "The name of the plan type supported by service."
-  default     = "standard"
+  description = "The cluster namespace where the Data Source Connector will be installed. Will be created if it does not exist."
+  default     = "data-source-connector"
+  nullable    = false
+}
+
+variable "dsc_registration_token" {
+  type        = string
+  description = "Registration token generated in the Backup & Recovery Service UI when adding a cluster data source."
+  sensitive   = true
+  nullable    = false
+}
+
+##############################################################################
+# Backup Recovery Service Instance
+##############################################################################
+
+variable "connection_id" {
+  type        = string
+  description = "Connection ID for the backup service"
+}
+
+variable "brs_instance_guid" {
+  type        = string
+  description = "GUID of the Backup & Recovery Service instance."
+  nullable    = false
+}
+
+variable "brs_instance_region" {
+  type        = string
+  description = "Region of the Backup & Recovery Service instance."
+  nullable    = false
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    condition     = contains(["us-east"], var.brs_instance_region)
+    error_message = "IKS/ROKS backup recovery is only supported in these regions: \"us-east\"."
   }
 }
 
-variable "resource_group_id" {
+variable "brs_endpoint_type" {
   type        = string
-  description = "The ID of the resource group where you want to create the service."
+  description = "The endpoint type to use when connecting to the Backup and Recovery service for creating a data source connection. Allowed values are 'public' or 'private'."
+  default     = "public"
+
+  validation {
+    condition     = contains(["public", "private"], var.brs_endpoint_type)
+    error_message = "`endpoint_type` must be 'public' or 'private'."
+  }
+}
+
+variable "brs_tenant_id" {
+  type        = string
+  description = "BRS tenant ID in the format `<tenant-guid>/`. Required for API calls and agent configuration."
+  nullable    = false
+}
+
+variable "registration_name" {
+  type        = string
+  description = "Name of the registration."
+  nullable    = false
 }
 
-variable "resource_tags" {
-  type        = list(string)
-  description = "List of resource tag to associate with the instance."
-  default     = []
+variable "registration_images" {
+  type = object({
+    data_mover              = optional(string, null)
+    velero                  = optional(string, null)
+    velero_aws_plugin       = optional(string, null)
+    velero_openshift_plugin = optional(string, null)
+    init_container          = optional(string, null)
```

Comment:
are these really all optional? I'm guessing they are required right? If so you should also add `nullable = false`